Core/CoreTests/Pages/PageList/PageListPresenterTests.swift

```diff
@@ -0,0 +1,129 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import XCTest
+@testable import Core
+import TestsFoundation
+
+class PageListPresenterTests: CoreTestCase {
+
+    var resultingError: NSError?
+    var resultingSubtitle: String?
+    var resultingBackgroundColor: UIColor?
+    var coursePresenter: PageListPresenter!
+    var groupPresenter: PageListPresenter!
+
+    let update = XCTestExpectation(description: "presenter updated")
+    var onUpdateNavBar: ((String?, UIColor?) -> Void)?
+    var onUpdate: () -> Void = {}
+
+    var color: UIColor?
+    var navigationController: UINavigationController?
+    var titleSubtitleView = TitleSubtitleView.create()
+    var navigationItem: UINavigationItem = UINavigationItem(title: "")
+
+    override func setUp() {
+        super.setUp()
+        coursePresenter = PageListPresenter(env: environment, view: self, context: ContextModel(.course, id: "42"))
+        groupPresenter = PageListPresenter(env: environment, view: self, context: ContextModel(.group, id: "42"))
+    }
+
+    func testLoadGroup() {
+        XCTAssertNil(resultingSubtitle)
+        XCTAssertNil(resultingBackgroundColor)
+
+        let g = Group.make(from: .make(id: "42"), in: databaseClient)
+        Color.make(canvasContextID: g.canvasContextID, color: UIColor.red)
+
+        let expectation = self.expectation(description: "navbar")
+        expectation.assertForOverFulfill = false
+        onUpdateNavBar = {
+            if $0 == g.name && $1 == g.color { expectation.fulfill() }
+        }
+        groupPresenter.viewIsReady()
+
+        wait(for: [expectation], timeout: 5)
+    }
+
+    func testLoadCourse() {
+        XCTAssertNil(resultingSubtitle)
+        XCTAssertNil(resultingBackgroundColor)
+
+        let c = Course.make(from: .make(id: "42"), in: databaseClient)
+        Color.make(canvasContextID: c.canvasContextID, color: UIColor.red)
+
+        let expectation = self.expectation(description: "navbar")
+        expectation.assertForOverFulfill = false
+        onUpdateNavBar = {
+            if $0 == c.name && $1 == c.color { expectation.fulfill() }
+        }
+        coursePresenter.viewIsReady()
+
+        wait(for: [expectation], timeout: 5)
+    }
+
+    func testLoadPages() {
+        Page.make(from: .make(title: "Answers Page"))
+        let expectation = self.expectation(description: "pages")
+        expectation.assertForOverFulfill = false
+        onUpdate = {
+            if !self.coursePresenter.pages.isEmpty {
+                expectation.fulfill()
+
+            }
+        }
+        coursePresenter.viewIsReady()
+        wait(for: [expectation], timeout: 1)
+        XCTAssertEqual(coursePresenter.pages.first?.title, "Answers Page")
```

Comment:
This test was flaky when it was in student. I think I fixed in in this PR https://github.com/instructure/canvas-ios/pull/472

Would you mind making that change here?